runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DataUtil.java

```diff
@@ -119,8 +119,8 @@ private static void serializePartition(final EncoderFactory encoderFactory,
         // We need to close wrappedStream on here, because DirectByteArrayOutputStream:getBufDirectly() returns
         // inner buffer directly, which can be an unfinished(not flushed) buffer.
         wrappedStream.close();
-        final byte[] serializedBytes = bytesOutputStream.getBufDirectly();
-        final int actualLength = bytesOutputStream.getCount();
+        final byte[] serializedBytes = bytesOutputStream.toByteArray();
```

Comment:
This will incur an array copy for the whole bytes.
Isn't there any other options (or any refactoring plan)?

Reply:
+1 for @sanha's comment regarding performance.

Reply:
I am thinking of using java.nio.ByteBuffer instead of ByteArrayOutputStream to avoid inefficiency. I will open a new PR for this thanks!